File: src/com/fabrica/muebles/modelo/produccion.py
from datetime import date


class Produccion:
    """
    Clase modelo que representa un registro de Producción
    Corresponde a la tabla: produccion

    Autor: William Alonso Samaca Lopez
    Sistema de Gestión Fábrica de Muebles
    """

    def __init__(self, id=None, codigo_mueble=None, nombre_cliente=None, tipo_documento_cliente='CC',
                 numero_documento_cliente=None, nombre_producto=None, cantidad=None,
                 fecha_inicio=None, fecha_fin=None, estado=None, observaciones=None, ruta_archivo=None):
        """
        Constructor de la clase Produccion

        Args:
            id: ID de la producción (autogenerado)
            codigo_mueble: Código del mueble asignado por producción
            nombre_cliente: Nombre del cliente
            tipo_documento_cliente: Tipo de documento del cliente (CC o NIT)
            numero_documento_cliente: Número de documento del cliente
            nombre_producto: Nombre del producto a fabricar
            cantidad: Cantidad a producir
            fecha_inicio: Fecha de inicio de producción
            fecha_fin: Fecha de finalización de producción
            estado: Estado de la producción
            observaciones: Observaciones del producto
            ruta_archivo: Ruta del archivo (foto/plano)
        """
        self.id = id
        self.codigo_mueble = codigo_mueble
        self.nombre_cliente = nombre_cliente
        self.tipo_documento_cliente = tipo_documento_cliente
        self.numero_documento_cliente = numero_documento_cliente
        self.nombre_producto = nombre_producto
        self.cantidad = cantidad
        self.fecha_inicio = fecha_inicio or date.today()
        self.fecha_fin = fecha_fin
        self.estado = estado
        self.observaciones = observaciones
        self.ruta_archivo = ruta_archivo

    def __str__(self):
        """Representación en string del objeto Produccion"""
        return (f"Produccion{{codigo='{self.codigo_mueble}', producto='{self.nombre_producto}', "
                f"cantidad={self.cantidad}, estado='{self.estado}'}}")

    def __repr__(self):
        return self.__str__()

File: modulo-java/src/com/fabrica/muebles/modelo/Produccion.java
package com.fabrica.muebles.modelo;

import java.sql.Date;

/**
 * Clase modelo que representa un registro de Producción
 * Corresponde a la tabla: produccion
 * 
 * Autor: William Alonso Samaca Lopez
 * Sistema de Gestión Fábrica de Muebles
 */
public class Produccion {
    
    // Atributos (coinciden con los campos de la tabla produccion)
    private int id;
    private String nombreProducto;
    private int cantidad;
    private Date fechaInicio;
    private Date fechaFin;
    private String estado;
    
    // Constructor vacío
    public Produccion() {
    }
    
    // Constructor para insertar (sin ID)
    public Produccion(String nombreProducto, int cantidad, Date fechaInicio, String estado) {
        this.nombreProducto = nombreProducto;
        this.cantidad = cantidad;
        this.fechaInicio = fechaInicio;
        this.estado = estado;
    }
    
    // Constructor completo (para consultas)
    public Produccion(int id, String nombreProducto, int cantidad, Date fechaInicio, 
                      Date fechaFin, String estado) {
        this.id = id;
        this.nombreProducto = nombreProducto;
        this.cantidad = cantidad;
        this.fechaInicio = fechaInicio;
        this.fechaFin = fechaFin;
        this.estado = estado;
    }
    
    // Getters y Setters
    public int getId() {
        return id;
    }
    
    public void setId(int id) {
        this.id = id;
    }
    
    public String getNombreProducto() {
        return nombreProducto;
    }
    
    public void setNombreProducto(String nombreProducto) {
        this.nombreProducto = nombreProducto;
    }
    
    public int getCantidad() {
        return cantidad;
    }
    
    public void setCantidad(int cantidad) {
        this.cantidad = cantidad;
    }
    
    public Date getFechaInicio() {
        return fechaInicio;
    }
    
    public void setFechaInicio(Date fechaInicio) {
        this.fechaInicio = fechaInicio;
    }
    
    public Date getFechaFin() {
        return fechaFin;
    }
    
    public void setFechaFin(Date fechaFin) {
        this.fechaFin = fechaFin;
    }
    
    public String getEstado() {
        return estado;
    }
    
    public void setEstado(String estado) {
        this.estado = estado;
    }
    
    @Override
    public String toString() {
        return "Produccion{" +
                "ID=" + id +
                ", producto='" + nombreProducto + '\'' +
                ", cantidad=" + cantidad +
                ", estado='" + estado + '\'' +
                '}';
    }
}